freqtrade/rpc/telegram.py

```diff
@@ -1097,6 +1099,208 @@ async def _profit(self, update: Update, context: CallbackContext) -> None:
+                )
+                markdown_msg = (
+                    "*ROI: Closed long trades*\n"
+                    f"∙ `{fmt_coin(profit_closed_coin, stake_cur)} "
+                    f"({profit_closed_ratio_mean:.2%}) "
+                    f"({profit_closed_percent} \u03A3%)`\n"
+                    f"{fiat_closed_trades}"
+                )
+            else:
+                markdown_msg = "`No closed long trade` \n"
+            fiat_all_trades = (
+                f"∙ `{fmt_coin(profit_all_fiat, fiat_disp_cur)}`\n" if fiat_disp_cur else ""
+            )
+            markdown_msg += (
+                f"*ROI: All long trades\n"
+                f"∙ `{fmt_coin(profit_all_coin, stake_cur)} "
+                f"({profit_all_ratio_mean:.2%}) "
+                f"({profit_all_percent} \u03A3%)`\n"
+                f"{fiat_all_trades}"
+                f"*Total Trade Count:* `{trade_count}`\n"
+                f"*Bot started:* `{stats['bot_start_date']}`\n"
+                f"*{'First Trade opened' if not timescale else 'Showing Profit since'}:* "
+                f"`{first_trade_date}`\n"
+                f"*Latest Trade opened:* `{latest_trade_date}`\n"
+                f"*Win / Loss:* `{stats['winning_trades']} / {stats['losing_trades']}`\n"
+                f"*Winrate:* `{winrate:.2%}`\n"
+                f"*Expectancy (Ratio):* `{expectancy:.2f} ({expectancy_ratio:.2f})`"
+            )
+            if stats["closed_trade_count"] > 0:
+                markdown_msg += (
+                    f"\n*Avg. Duration:* `{avg_duration}`\n"
+                    f"*Best Performing:* `{best_pair}: {best_pair_profit_abs} "
+                    f"({best_pair_profit_ratio:.2%})`\n"
+                    f"*Trading volume:* `{fmt_coin(stats['trading_volume'], stake_cur)}`\n"
+                    f"*Profit factor:* `{stats['profit_factor']:.2f}`\n"
+                    f"*Max Drawdown:* `{stats['max_drawdown']:.2%} "
+                    f"({fmt_coin(stats['max_drawdown_abs'], stake_cur)})`\n"
+                    f"    from `{stats['max_drawdown_start']} "
+                    f"({fmt_coin(stats['drawdown_high'], stake_cur)})`\n"
+                    f"    to `{stats['max_drawdown_end']} "
+                    f"({fmt_coin(stats['drawdown_low'], stake_cur)})`\n"
+                    f"*Current Drawdown:* `{stats['current_drawdown']:.2%} "
+                    f"({fmt_coin(stats['current_drawdown_abs'], stake_cur)})`\n"
+                    f"    from `{stats['current_drawdown_start']} "
+                    f"({fmt_coin(stats['current_drawdown_high'], stake_cur)})`\n"
+                )
+        await self._send_msg(
+            markdown_msg,
+            reload_able=True,
+            callback_path="update_profit_long",
+            query=update.callback_query,
+        )
+
+    @authorized_only
+    async def _profit_short(self, update: Update, context: CallbackContext) -> None:
```

Comment:
I'd assume that all 3 funcitions are about 90% the same (i didn't check, but at a glance, it'd seem like that).

so right now it's like 3x100 lines of almost identical code.

it might make more sense to have a `_format_profit_message(stats, side)` (or not sure on parameters, maybe pass update / context in), or even use `functools.partial` to have one method with the side as additional keyword argument to reduce code duplication? (which will inevitably drift apart with every future change).